### ISSUE TITLE
TaintedChecker false positive with Vector<Object> and .copyInto

### ISSUE THREAD

**Issue by shuehner** [typetools/checker-framework#2422]
Using TaintingChecker (i.e. 2.8.0) on code below reports a false positive for the copyInto call.

https://gist.github.com/shuehner/de4a2d6a551fceb60e10a5d279508272

[jsr308.javac] /home/huehner/ob/branches/pi_checker_pg/src/SHUTest1.java:8: error: [vector.copyinto.type.incompatible] incompatible component type in Vector.copyinto.
[jsr308.javac]          vector.copyInto(objectListData);
[jsr308.javac]                          ^
[jsr308.javac]   found   : @Tainted Object
[jsr308.javac]   required: @Tainted SHUTest1

Note that his only happens when the Vector is typed as 'Vector<Object>' intead of type matching the argument of copyInto.

While using <Object> here does not make any sense can be found in the wild of big / old-codebase.

Note: This could be related / duplicate of either #104 or #660 based on their descriptions.

**Comment by wmdietl** [comment]
This is the intended behavior, but it needs to be documented in the manual.
The motivation is documented in https://github.com/typetools/checker-framework/blob/master/framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java#L1464:

```
     * <p>The Checker Framework special-cases the method invocation, as its type safety cannot be
     * expressed by Java's type system.
     *
     * <p>For a Vector {@code v} of type {@code Vector<E>}, the method invocation {@code
     * v.copyInto(arr)} is type-safe iff {@code arr} is an array of type {@code T[]}, where {@code
     * T} is a subtype of {@code E}.
     *
     * <p>In other words, this method checks that the type argument of the receiver method is a
     * subtype of the component type of the passed array argument.
```

I don't think this is related to #104 or #660.

**Comment by wmdietl** [comment]
This is related to invariant arrays, as discussed here: https://checkerframework.org/manual/#invariant-arrays
Maybe this conservative behavior for `copyInto` should also be guarded by `-AinvariantArrays`.
